### ISSUE TITLE
gcd(10^10,5) returns undef

### ISSUE THREAD

**Issue by parisseb** [numworks/epsilon#75]
It can be fixed by using long long instead of int (overflow happens for integers larger than 2^53 instead of 2^31), with the following code in poincare/src/great_common_divisor.cpp
```cpp
template<typename T>
Evaluation<T> * GreatCommonDivisor::templatedEvaluate(Context& context, AngleUnit angleUnit) const {
  Evaluation<T> * f1Input = m_args[0]->evaluate<T>(context, angleUnit);
  Evaluation<T> * f2Input = m_args[1]->evaluate<T>(context, angleUnit);
  T f1 = f1Input->toScalar();
  T f2 = f2Input->toScalar();
  delete f1Input;
  delete f2Input;
  long long upper=(1LL<<53);
  if (f1>upper || -f1>upper || f2>upper || -f2>upper || isnan(f1) || isnan(f2) || f1 != (long long)f1 || f2 != (long long)f2) {
    return new Complex<T>(Complex<T>::Float(NAN));
  }
  long long a = f2,b = f1,r;
  while(b!=0){
    r = a -std::floor(double(a)/b)* b; // a%b and a/b not supported on ARM
    a = b;
    b = r;
  }
  if (a<0) a=-a; // insure positive gcd if b=0
  return new Complex<T>(Complex<T>::Float(std::round((T)a)));
}
```

**Comment by adriweb** [comment]
_(You can use ` ```cpp ` before and ` ``` ` after the code to make it code-blocked and highlighted :))_

**Comment by artaxxx** [comment]
Fixed in the last version 1.2.0
